python/BioSimSpace/Sandpit/Exscientia/Align/_alch_ion.py

```diff
@@ -0,0 +1,53 @@
+from .._SireWrappers import Molecule as _Molecule
+from .._Exceptions import IncompatibleError as _IncompatibleError
+
+
+def _mark_alchemical_ion(molecule):
+    """
+    Mark the ion molecule as being alchemical ion.
+
+    This enables one to use
+
+        * :meth:`~BioSimSpace.Sandpit.Exscientia._SireWrappers._system.System.getAlchemicalIon` to get the alchemical ion.
+        * :meth:`~BioSimSpace.Sandpit.Exscientia._SireWrappers._system.System.getAlchemicalIonIdx` to get the index of alchemical ion.
+        * :meth:`~BioSimSpace.Sandpit.Exscientia._SireWrappers._molecule.Molecule.isAlchemicalIon` to check if a molecule is an alchemical ion.
+
+
+    Parameters
+    ----------
+
+    molecule : BioSimSpace._SireWrappers.Molecule
+        The molecule to be marked as alchemical ion.
+
+    Returns
+    -------
+
+    alchemical_ion : BSS._SireWrappers.Molecule
+        The molecule marked as being alchemical ion.
+    """
+    # Validate input.
+
+    if not isinstance(molecule, _Molecule):
+        raise TypeError(
+            "'molecule' must be of type 'BioSimSpace._SireWrappers.Molecule'"
+        )
+
+    # Cannot decouple a perturbable molecule.
+    if molecule.isAlchemicalIon():
+        raise _IncompatibleError(
```

Comment:
I am wondering if this should be raising an error or just a warning instead

Reply:
This means that you are remarking an alchemical ion as an alchemical ion?
Currently this step is only executed in `fix_perturbed_charge` in BSS-wrappers so I think this should be a error?
I cannot think of a use case where we want to tolerate this thing?

Reply:
I don't know, I am not sure idempotent operations should be raising an error. Imagine you are writing some code that makes sure some dodgy input is correctly marked. Then on top of calling this function you'd have to do error handling every single time... And I don't really see the benefit of that.